internal/node/test/BUILD.bazel

```diff
@@ -604,3 +604,102 @@ nodejs_test(
         "@bazel_tools//src/conditions:windows": "@node16_windows_amd64//:node_toolchain",
     }),
 )
+
+# these rules create different targets for each toolchain that is supplied in the list for toolchains
+# this makes it easy to build or test against multiple versions of a toolchain at the same time with less duplicate work
+
+# this will create 2 nodejs_binary targets
+# bazel build //internal/node/test:main_multiple_loop_0
+# bazel build //internal/node/test:main_multiple_loop_1
+# the numbers are based on the order that the toolchains are passed to the rule
+[
+    nodejs_binary(
+        name = "main_multiple_loop_" + str(id),
+        entry_point = "binary_version.js",
+        node = tc,
+        toolchain = tc,
+    )
+    for id, tc in enumerate([
+        # using the select statement will download toolchains for all platforms
+        # you can also just provide an individual toolchain if you don't want to download them all
+        select({
+            "@bazel_tools//src/conditions:linux_x86_64": "@node16_linux_amd64//:node_toolchain",
+            "@bazel_tools//src/conditions:darwin": "@node16_darwin_amd64//:node_toolchain",
+            "@bazel_tools//src/conditions:windows": "@node16_windows_amd64//:node_toolchain",
+        }),
+        select({
+            "@bazel_tools//src/conditions:linux_x86_64": "@node15_linux_amd64//:node_toolchain",
+            "@bazel_tools//src/conditions:darwin": "@node15_darwin_amd64//:node_toolchain",
+            "@bazel_tools//src/conditions:windows": "@node15_windows_amd64//:node_toolchain",
+        }),
+    ])
+]
+
+# this will create 2 nodejs_binary targets
+# bazel build //internal/node/test:main_multiple_loop_node16
+# bazel build //internal/node/test:main_multiple_loop_node15
+# the numbers are based on the order that the toolchains are passed to the rule
+[
+    nodejs_binary(
+        name = "main_multiple_loop_" + id,
+        entry_point = "binary_version.js",
+        node = tc,
```

Comment:
Are you aiming to collapse node & toolchain into just one attr?

Reply:
I was hoping to get the use of toolchains added and then go back and clean this up. Originally it was behind a macro as well so a change collapsing the two wouldn't have been visible. But if we want to combine them now we could do that as well

Reply:
A follow-up of removing `node` in favor `toolchain` sounds good to me. Seems easier to do it that way and the downstream effects will be easier to see.